tests/e2e/users/add-user.wdio-spec.js

```diff
@@ -0,0 +1,105 @@
+const utils = require('../../utils');
+const faker = require('faker');
+const chai = require('chai');
+const usersAdminPage = require('../../page-objects/admin/user.wdio.page');
+const placeFactory = require('../../factories/cht/contacts/place');
+const loginPage = require('../../page-objects/login/login.wdio.page');
+const personFactory = require('../../factories/cht/contacts/person');
+
+const onlineUserRole = 'Program Officer';
+const offlineUserRole = 'CHW';
+const username = faker.name.firstName().toLocaleLowerCase();
+const password = 'Jacktest@123';
+const incorrectpassword = 'Passwor';
+const places = placeFactory.generateHierarchy();
+const district_hospital = places.find((place) => place.type === 'district_hospital');
+
+const person = personFactory.build(
+  {
+    parent: {
+      _id: district_hospital._id,
+      parent: district_hospital.parent
+    }
+  });
+
+const docs = [...places, person];
+
+describe('User Test Cases ->', () => {
+
+  before(async () => {
+    await utils.saveDocs(docs);
+    await loginPage.cookieLogin();
+  });
+
+  after(async () => {
+    await utils.revertDb([], true);
+  });
+
+  beforeEach(async () => {
+    await usersAdminPage.goToAdminUser();
+    await usersAdminPage.openAddUserDialog();
+  });
+
+  describe('Creating Users ->', () => {
+
+    it('should add user with valid password', async () => {
+      await usersAdminPage.inputAddUserFields(username, 'Jack', onlineUserRole, district_hospital.name, 
+        person.name, password);
+      await usersAdminPage.saveUser();
+      chai.expect(await usersAdminPage.getAllUsernames()).to.contain.members([username]);
+      await utils.deleteUsers([{ username: username }]);
```

Comment:
If the above `expect` fails, this line won't run and the users won't be deleted, which may bring undesired side effects, if not now, possibly in the future as things continue to grow.

To avoid this, and use the testing framework in a more idiomatic way, please move this kind of cleanup code into an `afterEach` function. One almost-valid objection to that might be for a case where that afterEach behavior would run, but for an `it` that doesn't need such cleanup. Take that as a sign that the `describe` setup is incorrect. Each `it` inside a `describe` should be checking a different variant of the same thing. Remember, you can nest `describe`s too as cases narrow. This instance of course is much simpler, having only a single test, but my intention here is to describe (hah!) the importance of settings these up thoughtfully and the path forward as logic and tests grow.

Reply:
Done